launch/lifelong_launch.py

```diff
@@ -1,17 +1,63 @@
-from launch import LaunchDescription
-import launch_ros.actions
 from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import (DeclareLaunchArgument, EmitEvent, LogInfo,
+                            RegisterEventHandler)
+from launch.conditions import IfCondition
+from launch.events import matches_action
+from launch.substitutions import LaunchConfiguration
+from launch_ros.actions import LifecycleNode
+from launch_ros.event_handlers import OnStateTransition
+from launch_ros.events.lifecycle import ChangeState
+from lifecycle_msgs.msg import Transition
 
 
 def generate_launch_description():
-    return LaunchDescription([
-        launch_ros.actions.Node(
+    autostart = LaunchConfiguration('autostart')
+
+    declare_autostart_cmd = DeclareLaunchArgument(
+        'autostart', default_value='true',
+        description='Automatically startup the slamtoolbox')
+
+    start_lifelong_slam_toolbox_node = LifecycleNode(
           parameters=[
             get_package_share_directory("slam_toolbox") + '/config/mapper_params_lifelong.yaml'
           ],
           package='slam_toolbox',
           executable='lifelong_slam_toolbox_node',
           name='slam_toolbox',
-          output='screen'
-        )
-    ])
+          output='screen',
+          namespace=''
```

Comment:
Again, remove the namespace field from all launch files

Reply:
<https://github.com/ros2/launch_ros/blob/humble/launch_ros/launch_ros/actions/lifecycle_node.py#L45>

`launch_ros.actions.LifeCycleNode` requires `namespace` that cannot be removed.

```diff
$ git diff
diff --git a/launch/online_sync_launch.py b/launch/online_sync_launch.py
index c2bfcd1..27cd866 100644
--- a/launch/online_sync_launch.py
+++ b/launch/online_sync_launch.py
@@ -40,7 +40,6 @@ def generate_launch_description():
         executable='sync_slam_toolbox_node',
         name='slam_toolbox',
         output='screen',
-        namespace=''
     )
 
     configure_event = EmitEvent(
```

```console
$ ros2 launch slam_toolbox online_sync_launch.py 
[INFO] [launch]: All log files can be found below /home/ubuntu/.ros/log/2023-02-07-22-58-10-364638-c6ea82625ecd-828
[INFO] [launch]: Default logging verbosity is set to INFO
[ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught exception when trying to load file of format [py]: LifecycleNode.__init__() missing 1 required keyword-only argument: 'namespace'
```

Reply:
Wow thats odd - OK!